plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notation/crowsfoot/CrowsFootDiagramNotation.java

```diff
@@ -115,6 +114,27 @@ private void createTargetDecorator(PolylineConnection conn, Color bckColor, Colo
         conn.setTargetDecoration(targetDecor);
     }
 
+    private List<DBSTableIndex> getIndexSourceAttributes(
+        Collection<? extends DBSTableIndex> indexes,
+        List<ERDEntityAttribute> sourceAttr,
+        DBRProgressMonitor monitor) throws DBException {
+        List<DBSTableIndex> sourceAttributeIndexes = new ArrayList<>();
+        List<DBSEntityAttribute> sourceAttributes = sourceAttr.stream().map(ERDEntityAttribute::getObject).toList();
+        for (DBSTableIndex index : indexes) {
+            if (index.isPrimary()) {
```

Comment:
Why test for primary index?
It is supported by very few number of databases.
If you want to exclude primary keys then you need to check PK constraint (not index)

Reply:
Removed extra condition